src/main.rs

```diff
@@ -142,9 +142,9 @@ where
     // startup online events in one go (before we boot up the consumer). The number of peers that we have is also
     // related to the number of events we can expect to process, so it's a sensible enough buffer size.
     let (sender, receiver) = channel(current_peers.len() + 1);
-    for (peer, onion_support) in current_peers {
+    for (peer, onion_support) in &current_peers {
         sender
-            .send(MessengerEvents::PeerConnected(peer, onion_support))
+            .send(MessengerEvents::PeerConnected(*peer, *onion_support))
```

Comment:
I turned that into a reference on line 145 so that it isn't "moved" on line 145, and so we can reference it in the call to consume_messenger_events on line 262. If helpful I can move that to another commit.

Reply:
(Removed this with most recent changes!)